File: JavaScript/547. Number of Provinces.js
/**
 * @param {number[][]} isConnected
 * @return {number}
 */
var findCircleNum = function(isConnected) {
    var n= isConnected.length;
    var sol=0;
    var vis = new Array(n).fill(0);
    for(var i=0;i<n;i++){
        if(vis[i]==0){
            sol++;
            dfs(isConnected, vis,i);
        }
    }
    return sol;
};
var dfs = function(isConnected, vis,j){
    var n = isConnected.length;
    vis[j]=1;
    for(var i=0;i<n;i++){
        if(i!=j && vis[i]==0 && isConnected[i][j]==1)
            dfs(isConnected, vis, i);
    }
}

File: Java/547. Number of Provinces.java
class Solution {
    public int findCircleNum(int[][] isConnected) {
        int n= isConnected.length;
        int sol=0;
        int[] vis = new int[n];
        for(int i=0;i<n;i++){
            if(vis[i]==0){
                sol++;
                dfs(isConnected, vis,i);
            }
        }
        return sol;
    }
    public void dfs(int[][] isConnected, int[] vis,int j){
        int n = isConnected.length;
        vis[j]=1;
        for(int i=0;i<n;i++){
            if(i!=j && vis[i]==0 && isConnected[i][j]==1)
                dfs(isConnected, vis, i);
        }
    }
}
